### PR TITLE
routing: Fix lots of 'message to uknown sysid' errors

### PR DESCRIPTION
**Pull request by edersondisouza**
mavlink2 messages may be trimmed - i.e., any trailing 0 byte is
discarded. However, mavlink-router wasn't taking that into consideration
when getting target_system from a message: it was reading past message
length.

This patch just assumes broadcast if target_system offset is past
message length: it was zero anyway.

### CODE REVIEW

File: endpoint.cpp [hunk 1]
```diff
@@ -236,8 +238,11 @@ int Endpoint::read_msg(struct buffer *pbuf, int *target_sysid)
     if (msg_entry == nullptr) {
         log_debug("No message entry for %u", msg_id);
     } else {
-        if (msg_entry->flags & MAV_MSG_ENTRY_FLAG_HAVE_TARGET_SYSTEM)
-            *target_sysid = payload[msg_entry->target_system_ofs];
+        if (msg_entry->flags & MAV_MSG_ENTRY_FLAG_HAVE_TARGET_SYSTEM) {
+            // if target_system is 0, it may have been trimmed out on mavlink2
```
Comment (lucasdemarchi) [inline endpoint.cpp]: so the message actually has a target_system. Shouldn't you set *target_sysid to 0 in this case? Then it's clear -1 means "no target sysid" and 0 means broadcast.

Comment (edersondisouza) [inline endpoint.cpp]: Yeah, it would be more semantically accurate.

Comment (zehortigoza) [inline endpoint.cpp]: +1

**Comment by zehortigoza** [comment]
LGTM just fix the typo on the commit message.

**Comment by edersondisouza** [comment]
Updated to broadcast `0` and fix typo commit message.

**Pull request merged**